a10_neutron_lbaas/a10_config.py

```diff
@@ -28,18 +28,29 @@
 
 
 class ConfigModule(object):
-    def __init__(self, d):
+    def __init__(self, d, provider=None):
         self.__dict__.update(d)
 
+        if provider is None or 'providers' not in d or provider not in d['providers']:
+            return None
+
+        for k, v in d['providers'][provider].items():
+            if isinstance(v, dict):
+                if k not in self.__dict__:
+                    self.__dict__[k] = {}
+                self.__dict__[k].update(v)
+            else:
+                self.__dict__[k] = v
+
```

Comment:
Yeah, so, does a passed in config mean "use *this* config", or "use this object as if it was the file config.py" ?